x-pack/plugins/apm/common/utils/formatters/formatters.ts

```diff
@@ -14,14 +15,13 @@ export function asInteger(value: number) {
   return numeral(value).format('0,0');
 }
 
-export function tpmUnit(type?: string) {
-  return type === 'request'
-    ? i18n.translate('xpack.apm.formatters.requestsPerMinLabel', {
-        defaultMessage: 'rpm',
-      })
-    : i18n.translate('xpack.apm.formatters.transactionsPerMinLabel', {
-        defaultMessage: 'tpm',
-      });
+export function tpmUnit(unit: DateBucketUnit) {
+  return i18n.translate('xpack.apm.formatters.transactionsRateLabel', {
+    defaultMessage: 'tp{unit, select, minute {m} second {s}}',
+    values: {
+      unit,
+    },
+  });
```

Comment:
I started doing this, but in most cases it's more readable to have it as a label. You end up with a ternary in ternaries in some other situations, or in a template string. Plus, if we add another unit (e.g. ms), this will be easier to change. Thoughts?

Reply:
If we have three (or more) cases I'd go for a switch or good old if/else:
```ts

switch(unit) {
  case "millisecond":
      return i18n.translate('xpack.apm.formatters.transactionsPerSecLabel', {
        defaultMessage: 'tpms',
      });
  case "second":
      return i18n.translate('xpack.apm.formatters.transactionsPerSecLabel', {
        defaultMessage: 'tps',
      });
  case "minute":
      i18n.translate('xpack.apm.formatters.transactionsPerMinLabel', {
        defaultMessage: 'tpm',
      });
}